pkg/api/fs.go

```diff
@@ -515,55 +515,82 @@ func (f *Fs) Update(stream pb.Fs_UpdateServer) error {
 				return status.Errorf(errStatus.Code(), "FS receive update request: %v", err)
 			}
 
-			if nextVersion == -1 {
-				if project == -1 {
-					project = req.Project
-				}
-
-				latestVersion, err = db.LockLatestVersion(ctx, tx, project)
-				if errors.Is(err, db.ErrNotFound) {
-					return status.Errorf(codes.NotFound, "FS update missing latest version: %v", err)
-				}
-				if err != nil {
-					return status.Errorf(codes.Internal, "FS update lock latest version: %v", err)
-				}
-
-				nextVersion = latestVersion + 1
-				logger.Debug(ctx, "FS.Update[Init]", key.Project.Field(project), key.Version.Field(nextVersion))
+			// project can be -1 if the client authenticated using an Admin token
+			if project == -1 {
+				project = req.Project
+			}
+			if project != req.Project {
+				return status.Errorf(codes.InvalidArgument, "initial project %v, next project %v: %v", project, req.Project, ErrMultipleProjectsPerUpdate)
+			}
 
+			// We can only create the pack manager once we have the project ID and that requires a least one stream message
+			if packManager == nil {
 				packManager, err = db.NewPackManager(ctx, tx, project)
 				if err != nil {
 					return status.Errorf(codes.Internal, "FS create packed cache: %v", err)
 				}
 
 				span.SetAttributes(
 					key.Project.Attribute(project),
-					key.Version.Attribute(nextVersion),
 				)
 			}
 
-			if project != req.Project {
-				return status.Errorf(codes.InvalidArgument, "initial project %v, next project %v: %v", project, req.Project, ErrMultipleProjectsPerUpdate)
-			}
-
 			packParent := packManager.IsPathPacked(req.Object.Path)
 			if packParent != nil {
-				buffer[*packParent] = append(buffer[*packParent], req.Object)
+				packedBuffer[*packParent] = append(packedBuffer[*packParent], req.Object)
 				continue
 			}
 
+			objectBuffer = append(objectBuffer, req.Object)
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return err
```

Comment:
I noticed below we do a rollback, but here we just return. Just checking in to see if that's intentional, or should we be rolling back?

Reply:
I think this ones fine because [this `close`](https://github.com/gadget-inc/dateilager/pull/59/files#diff-4e519844ce517f4d280edd619be25f5692d32e2243d090437200301f0281b17aR497) will automatically rollback the transaction.